File: C++/String/415. Add Strings.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        string ans = "";
        int i = num1.size() - 1, j = num2.size() - 1;
        int carry = 0;
        while (i >= 0 || j >= 0 || carry != 0) {
            char c1 = i>=0 ? num1[i] : '0';
            char c2 = j>=0 ? num2[j] : '0';
            int sum = (c1 - '0') + (c2 - '0') + carry;
            int digit = sum % 10;
            carry = sum / 10;
            ans += to_string(digit);
            i--;
            j--;
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: Python/Archive/415.add_strings.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        int1 = 0
        for c in num1:
            int1 = int1 * 10 + ord(c) - ord('0')
        int2 = 0
        for c in num2:
            int2 = int2 * 10 + ord(c) - ord('0')
        
        return str(int1 + int2)